src/Lib.P2P/Cid.cs

```diff
@@ -0,0 +1,584 @@
+        ///   Returns a string representation of the CID
+        ///   according to the provided format specifier.
+        /// </summary>
+        /// <param name="format">
+        ///   A single format specifier that indicates how to format the value of the
+        ///   CID.  Can be "G" or "L".
+        /// </param>
+        /// <returns>
+        ///   The CID in the specified <paramref name="format"/>.
+        /// </returns>
+        /// <exception cref="FormatException">
+        ///   <paramref name="format"/> is not valid.
+        /// </exception>
+        /// <remarks>
+        /// <para>
+        ///   The "G" format specifier is the same as calling <see cref="Encode"/>.
+        /// </para>
+        /// <list type="table">  
+        /// <listheader>  
+        ///   <term>Specifier</term>  
+        ///   <description>return value</description>  
+        /// </listheader>  
+        ///  <item>  
+        ///    <term>G</term>  
+        ///    <description>QmXg9Pp2ytZ14xgmQjYEiHjVjMFXzCVVEcRTWJBmLgR39V</description>  
+        ///  </item>  
+        ///  <item>  
+        ///    <term>L</term>  
+        ///    <description>base58btc cidv0 dag-pb sha2-256 Qm...</description>  
+        ///  </item>  
+        /// </list> 
+        /// </remarks>
+        public string ToString(string format)
+        {
+            switch (format)
+            {
+                case "G":
+                    return Encode();
+
+                case "L":
+                    var sb = new StringBuilder();
+                    sb.Append(Encoding);
+                    sb.Append(' ');
+                    sb.Append("cidv");
+                    sb.Append(Version);
+                    sb.Append(' ');
+                    sb.Append(ContentType);
+                    if (Hash != null)
+                    {
+                        sb.Append(' ');
+                        sb.Append(Hash.Algorithm.Name);
+                        sb.Append(' ');
+                        sb.Append(MultiBase.Encode(Hash.ToArray(), Encoding).Substring(1));
+                    }
+
+                    return sb.ToString();
+
+                default:
+                    throw new FormatException($"Invalid CID format specifier '{format}'.");
+            }
+        }
+
+        /// <summary>
+        ///   Converts the CID to its equivalent string representation.
+        /// </summary>
+        /// <returns>
+        ///   The string representation of the <see cref="Cid"/>.
+        /// </returns>
+        /// <remarks>
+        ///   For <see cref="Version"/> 0, this is equalivalent to the 
+        ///   <see cref="MultiHash.ToBase58()">base58btc encoding</see>
+        ///   of the <see cref="Hash"/>.
+        /// </remarks>
+        /// <seealso cref="Decode"/>
+        public string Encode()
+        {
+            if (encodedValue != null) return encodedValue;
+            if (Version == 0)
+                encodedValue = Hash.ToBase58();
+            else
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'else' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)